### ISSUE TITLE
Unable to Commit Block on Testnet

### ISSUE THREAD

**Issue by Blackglade** [algorand/indexer#693]
### Subject of the issue

Just upgraded to Indexer v2.7.1 and Algorand v3.0.1 on my mainnet, testnet, and betanet nodes. Having trouble with my testnet indexer. Everytime I try to turn it on, I run into this issue:

```
{"level":"info","msg":"serving on :8980","time":"2021-09-30T00:25:50Z"}
{"level":"info","msg":"Running 0 migrations.","time":"2021-09-30T00:25:50Z"}
{"level":"info","msg":"Setting status: Migrations Complete","time":"2021-09-30T00:25:50Z"}
{"level":"info","msg":"Migration finished successfully.","time":"2021-09-30T00:25:50Z"}
⇨ http server started on [::]:8980
{"level":"info","msg":"Initializing block import handler.","time":"2021-09-30T00:25:50Z"}
{"level":"info","msg":"Starting block importer.","time":"2021-09-30T00:25:50Z"}
{"level":"info","msg":"adding block 16962372","time":"2021-09-30T00:25:50Z"}
{"error":"AddBlock() eval err: transaction TSGJV33ECSNLF6TRHUI2X26B6ZQVHMZYQFHAEZ5BB6NIMPBIVFOA: logic eval error: - would result negative","level":"error","msg":"adding block 16962372 to database failed","time":"2021-09-30T00:25:50Z"}
```

No idea why it's unable to commit this transaction on testnet while the indexer is working flawlessly on Mainnet/Betanet. Not sure if this is a bug or something on my end. 

### Your environment

```
12884901889
3.0.1.stable [rel/stable] (commit #b619b940)
go-algorand is licensed with AGPLv3.0
source code available at https://github.com/algorand/go-algorand

Running on Debian 10 ARM Architecture
```

**Comment by bert2002** [comment]
Hi, we have the same issue. Just upgraded and having this on testnet:

```
Sep 30 03:15:42 algorand-1 systemd[1]: Started Algorand Indexer daemon.
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: Using configuration file: /var/lib/algorand/indexer.yml
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: serving on xxx:8980
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"serving on xxx:8980","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"Running 0 migrations.","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"Setting status: Migrations Complete","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"Migration finished successfully.","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: ⇨ http server started on xxx:8980
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"Initializing block import handler.","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"Starting block importer.","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"level":"info","msg":"adding block 16964981","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 algorand-indexer[27645]: {"error":"AddBlock() eval err: transaction S7I3F67VVBF6CAJR2FK3XWPFWZ7EPLZBKYEV63K2QH3WCHOQZVLQ: logic eval error: - would result negative","level":"error","msg":"adding block 16964981 to database failed","time":"2021-09-30T03:15:42Z"}
Sep 30 03:15:42 algorand-1 systemd[1]: algorand-indexer.service: Main process exited, code=exited, status=1/FAILURE
Sep 30 03:15:42 algorand-1 systemd[1]: algorand-indexer.service: Failed with result 'exit-code'.

```

**Comment by Blackglade** [comment]
Downgrading back down to 2.6.2 resolves this. Definitely an issue with 2.7.1

**Comment by winder** [comment]
This is resolved in 2.8.0, thank you for the report.
